Foundation/NSURL.swift

```diff
@@ -858,9 +858,49 @@ public class NSURLComponents: NSObject, NSCopying {
     public override func copy() -> AnyObject {
         return copy(with: nil)
     }
-    
+
+    public override func isEqual(_ object: AnyObject?) -> Bool {
+        if let other = object as? NSURLComponents {
+            if scheme != other.scheme {
+                return false
+            }
+            if user != other.user {
+                return false
+            }
+            if password != other.password {
+                return false
+            }
+            if host != other.host {
+                return false
+            }
+            if port != other.port {
+                return false
+            }
+            if path != other.path {
+                return false
+            }
+            if query != other.query {
+                return false
+            }
+            if fragment != other.fragment {
+                return false
+            }
+            return true
+        }
+        return false
+    }
```

Comment:
CFURLComponentsRef implements an equality checking function. Can we just use CFEqual?